scripts/travis-emscripten/publish_binary.sh

```diff
@@ -34,7 +34,10 @@ VER=$(cat CMakeLists.txt | grep 'set(PROJECT_VERSION' | sed -e 's/.*set(PROJECT_
 test -n "$VER"
 VER="v$VER"
 COMMIT=$(git rev-parse --short HEAD)
-DATE=$(date --date="$(git log -1 --date=iso --format=%ad HEAD)" --utc +%Y.%m.%d)
+DATE=$(date --date="$(git log -1 --date=iso --format=%ad HEAD)" --utc +%Y.%-m.%-d)
+
+# remove leading zeros in components - they are not semver-compatible
+COMMIT=$(echo "$COMMIT" | sed -e 's/^0*//')
```

Comment:
Was it wrong to add this here or was this part intentional as well? This is basically what introduces https://github.com/ethereum/solidity/issues/8127

Reply:
It was intentional. There cannot be leading zeros in this semver component.